controller.go

```diff
@@ -43,24 +43,12 @@ func SyncPVC(pvc *PVClaim) {
 				// No PV could be found
 				// OBSERVATION: pvc is "Pending", will retry
 				if hasAnnotation(pvc, annClass) {
-					plugin := findProvisionerPluginForPV(pv) // Need to flesh this out
-					if plugin != nil {
-						//FIXME: left off here
-						// No match was found and provisioning was requested.
-						//
-						// maintain a map with the current provisioner goroutines that are running
-						// if the key is already present in the map, return
-						//
-						// launch the goroutine that:
-						// 1. calls plugin.Provision to make the storage asset
-						// 2. gets back a PV object (partially filled)
-						// 3. create the PV API object, with claimRef -> pvc
-						// 4. deletes itself from the map when it's done
-						// return
-					} else {
-						// make an event calling out that no provisioner was configured
-						// return, try later?
+					if err = provisionClaim(claim); err != nil {
+						return err
 					}
+					// Provisioning in progress, do not continue processing this
+					// claim.
+					return nil
```

Comment:
since this code is gradually morphing closer to real go, would you rather:

1.  simply return here (this func does not have a return type currently)
2.  add a return type of `error` and add appropriate returns throughout the rest of the method

I know your actual golang refactor of the code is in progress but I want to keep the pseudocode internally consistent and easy to reason about.